src/fluence/fluxion/fluxion.go

```diff
@@ -54,42 +56,80 @@ func (s *Fluxion) Cancel(ctx context.Context, in *pb.CancelRequest) (*pb.CancelR
 	// Why would we have an error code here if we check above?
 	// This (I think) should be an error code for the specific job
 	dr := &pb.CancelResponse{JobID: in.JobID}
-	fmt.Printf("[GRPCServer] Sending Cancel response %v\n", dr)
-
-	fmt.Printf("[CancelRPC] Errors so far: %s\n", s.cli.GetErrMsg())
+	klog.Infof("[Fluence] sending cancel response %v\n", dr)
+	klog.Infof("[Fluence] cancel errors so far: %s\n", s.cli.GetErrMsg())
 
 	reserved, at, overhead, mode, fluxerr := s.cli.Info(int64(in.JobID))
-	fmt.Println("\n\t----Job Info output---")
-	fmt.Printf("jobid: %d\nreserved: %t\nat: %d\noverhead: %f\nmode: %s\nerror: %d\n", in.JobID, reserved, at, overhead, mode, fluxerr)
+	klog.Infof("\n\t----Job Info output---")
+	klog.Infof("jobid: %d\nreserved: %t\nat: %d\noverhead: %f\nmode: %s\nerror: %d\n", in.JobID, reserved, at, overhead, mode, fluxerr)
 
-	fmt.Printf("[GRPCServer] Sending Cancel response %v\n", dr)
+	klog.Infof("[GRPCServer] Sending Cancel response %v\n", dr)
 	return dr, nil
 }
 
+// generateJobSpec generates a jobspec for a match request and returns the string
+func (s *Fluxion) generateJobspec(in *pb.MatchRequest) ([]byte, error) {
+
+	spec := []byte{}
+
+	// Create a temporary file to write and read the jobspec
+	// The first parameter here as the empty string creates in /tmp
+	file, err := os.CreateTemp("", "jobspec.*.yaml")
```

Comment:
Is it worth creating a temporary file just to turn it back into a string?

Reply:
Absolutely not! I refactored this in another variant of this API, but (when I was working on it here) I didn't have enough confidence to question the original design decision. I thought there must be a reason, so I left it. I'll try removing it (it simplifies the code a ton and also we don't write a file we don't need) and see if tests pass (or if there is something I'm missing).